File: Week 6/sentimental-credit/credit.py
from cs50 import get_int


def main():

    ccn = get_int("What is your credit card number?\n")
    ccntemp = ccn

    count = len(str(ccn))

    total_sum = 0

    for i in range(count):
        current = ccntemp % 10
        if i % 2 != 0:
            current *= 2
            if current > 9:
                current = current % 10 + current // 10
        total_sum += current
        ccntemp //= 10

    ccntemp = ccn
    while ccntemp >= 100:
        ccntemp //= 10

    first_digit = ccntemp // 10
    second_digit = ccntemp % 10

    if total_sum % 10 == 0:
        if first_digit == 4 and count in [13, 16]:
            print("VISA")
        elif first_digit == 5 and 1 <= second_digit <= 5 and count == 16:
            print("MASTERCARD")
        elif first_digit == 3 and second_digit in [4, 7] and count == 15:
            print("AMEX")
        else:
            print("INVALID")
    else:
        print("INVALID")


if __name__ == "__main__":
    main()

File: Week 1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    // User inputs their Credit Card Number - CCN

    long ccn = get_long("What is your credit card number?\n");
    long number = ccn;
    long ccnsaved = ccn;
    long ccntemp = ccn;

    // Counting the digits to get a digit count

    int count = 0;
    while (number != 0)
    {
        number = number / 10;
        count++;
    }

    // Luhn's algorithm

    int current;
    int total_sum = 0;

    for (int i = 0; i < count; i++)
    {
        current = ccn % 10;
        if (i % 2 != 0)
        {
            current = current * 2;
            if (current > 9)
            {
                current = current % 10 + current / 10;
            }
        }
        total_sum = total_sum + current;
        ccn = ccn / 10;
    }

    // Finding the card type

    int first_digit;
    int second_digit;
    if (total_sum % 10 == 0)
    {
        do
        {
            ccntemp = ccntemp / 10;
        }
        while (ccntemp >= 100);

        first_digit = ccntemp / 10;
        second_digit = ccntemp % 10;

        if ((first_digit == 4) && (count == 13 || count == 16))
        {
            printf("VISA\n");
        }

        else if ((first_digit == 5) && (second_digit >= 1) && (second_digit <= 5) && (count == 16))
        {
            printf("MASTERCARD\n");
        }

        else if ((first_digit == 3) && ((second_digit == 4) || (second_digit == 7)) &&
                 (count == 15))
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }

    else
    {
        printf("INVALID\n");
    }
}
